registry/storage/driver/s3-aws/s3.go

```diff
@@ -667,13 +667,13 @@ func (d *driver) List(ctx context.Context, opath string) ([]string, error) {
 			directories = append(directories, strings.Replace(commonPrefix[0:len(commonPrefix)-1], d.s3Path(""), prefix, 1))
 		}
 
-		if *resp.IsTruncated {
-			resp, err = d.S3.ListObjects(&s3.ListObjectsInput{
-				Bucket:    aws.String(d.Bucket),
-				Prefix:    aws.String(d.s3Path(path)),
-				Delimiter: aws.String("/"),
-				MaxKeys:   aws.Int64(listMax),
-				Marker:    resp.NextMarker,
+		if resp.IsTruncated != nil && *resp.IsTruncated {
```

Comment:
`&& *resp.IsTruncated` is not required

Reply:
I think it's better to keep it, considering that S3 compatible servers may return false.